jvm/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -287,6 +287,27 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
         case plan: CoalesceExec =>
           val transformer = CoalesceExecTransformer(plan.numPartitions, plan.child)
           TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+        case plan: GlobalLimitExec =>
+          if (!enableColumnarLimit) {
+            TransformHints.tagNotTransformable(plan)
+          } else {
+            val transformer = LimitTransformer(plan.child, 0L, plan.limit)
+            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+          }
+        case plan: LocalLimitExec =>
+          if (!enableColumnarLimit) {
+            TransformHints.tagNotTransformable(plan)
+          } else {
+            val transformer = LimitTransformer(plan.child, 0L, plan.limit)
+            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+          }
+        case plan: CollectLimitExec =>
+          if (!enableColumnarLimit) {
+            TransformHints.tagNotTransformable(plan)
+          } else {
+            val transformer = LimitTransformer(plan.child, 0L, plan.limit)
+            TransformHints.tag(plan, transformer.doValidate().toTransformHint)
+          }
```

Comment:
Is there any difference in the implementations of the three limit operators? cc @PHILO-HE

Reply:
I think the implementation of the CollectLimitExec is wrong, it will add a SinglePartition shuffle, but the vanilla spark will not.

Reply:
There is a single partiton shuffle in `CollectLimitExec`